### PR TITLE
select fields of categories and subjects on Subjects step for a tutor stepper #156

### LINKED ISSUE #156: (SP: 3 ) Add photo and select on Subjects step
UserStory: #75 

## Description
Add photo and select **fields** of categories and subjects on Subjects step for a tutor stepper

## Design 

Image

## Acceptance Criteria
- [ ] When user clicks on field with categories they should be fetched from server
- [ ] When user chose a category and clicked on subjects select it should fetch all subjects of this category
- [ ] When user clicks on criss-cross of select it should clear it

### PR DESCRIPTION
**Pull request by marianamartsy**
close #156 

Added photo and select fields of categories and subjects on Subjects step for a tutor stepper
- [x] When user clicks on field with categories they should be fetched from server
- [x] When user chose a category and clicked on subjects select it should fetch all subjects of this category
- [x] When user clicks on criss-cross of select it should clear it
Zrzut ekranu 2023-12-8 o 02 54 31
Zrzut ekranu 2023-12-8 o 02 54 41
Zrzut ekranu 2023-12-8 o 02 55 00

### CODE REVIEW

**Review by ArtemHolikov — APPROVED** [review]
👍

**Review by Mygal-Alex — APPROVED** [review]
👍

**Review by LeonidShamarin — APPROVED** [review]

**Review by AndriiNasadiuk — APPROVED** [review]
👌

File: src/containers/tutor-home-page/subjects-step/SubjectsStep.jsx [hunk 1]
```diff
@@ -1,13 +1,107 @@
...
+
+  const fetchCategories = useCallback(async () => {
+    if (categories.length > 0) {
+      return
+    }
+    try {
+      const response = await categoryService.getCategoriesNames()
+      setCategories(response.data)
+    } catch (error) {
+      console.error('Error fetching categories:', error)
```
Comment (vlad-khrychov) [inline src/containers/tutor-home-page/subjects-step/SubjectsStep.jsx:26]: Let`s add additional task to show this error not in console but for user.

**Review by vlad-khrychov — APPROVED** [review]
It is amazing work with tests.

File: src/containers/tutor-home-page/subjects-step/SubjectsStep.jsx (hunk 1 above)
Comment (marianamartsy) [reply src/containers/tutor-home-page/subjects-step/SubjectsStep.jsx:26]: 👍 i will add this new task to the board

**Pull request merged**